### PR TITLE
Extract `AndroidStudioProvisioningPlugin` from `PerformanceTestPlugin`

### LINKED ISSUE #27499: Extract AS provisioning logic from `performance-test-plugin`

### PR DESCRIPTION
**Pull request by 6hundreds**
Fixes #27499 

We want to reuse PerformanceTest AndroidStudio provisioning infrastructure in a new type of integration tests in CC project. These tests are going to test sync with AndroidStudio (and IDEA in future), so it need to obtain an IDE somehow. 

This PR is extracting logic of AS provisioning from `PerformanceTestPlugin` to `AndroidStudioProvisioningPlugin` as is.
Also, `AndroidStudioProvisioningPlugin` is allowing to configure a version of AndroidStudio that will be provided.

### CODE REVIEW

**Comment by 6hundreds** [comment]
@bot-gradle help

**Comment by 6hundreds** [comment]
@bot-gradle test PT

**Review by Michnia34 — APPROVED** [review]

File: build-logic/performance-testing/src/main/kotlin/gradlebuild/performance/PerformanceTestPlugin.kt [hunk 1]
```diff
@@ -99,8 +84,8 @@ object Config {
     const val performanceTestResultsJson = "performance-tests/$performanceTestResultsJsonName"
 
     // Android Studio Iguana 2023.2.1.16 Canary 16
-    const val androidStudioVersion = "2023.2.1.16"
-    val defaultAndroidStudioJvmArgs = listOf("-Xms256m", "-Xmx4096m")
+    const val performanceTestAndroidStudioVersion = "2023.2.1.16"
```
[the remark this answers is not in the document]
Comment (6hundreds) [reply build-logic/performance-testing/src/main/kotlin/gradlebuild/performance/PerformanceTestPlugin.kt:87]: We can do it, but these versions are not intended to be same in general. Project that applies plugin should configure desired version of AS and if no version is provided than default one is using.

File: build-logic/integration-testing/src/main/kotlin/gradlebuild/integrationtests/ide/AndroidStudioSystemProperties.kt [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    override fun asArguments(): Iterable<String> {
+        val systemProperties = mutableListOf<String>()
+        if (autoDownloadAndroidStudio) {
+            val androidStudioPath = studioInstallation.studioInstallLocation.asFile.get().absolutePath
+            systemProperties.add("-Dstudio.home=$androidStudioPath")
+        } else {
+            if (androidStudioHome.isPresent) {
+                systemProperties.add("-Dstudio.home=${androidStudioHome.get()}")
+            }
+        }
```
[the remark this answers is not in the document]
Comment (6hundreds) [reply build-logic/integration-testing/src/main/kotlin/gradlebuild/integrationtests/ide/AndroidStudioSystemProperties.kt]: @blindpirate Fixed

**Comment by 6hundreds** [comment]
@bot-gradle merge

**Pull request merged**